core/app/[locale]/(default)/product/[slug]/page.tsx

```diff
@@ -283,6 +283,12 @@ export default async function Product(props: Props) {
     };
   });
 
+  // Fire-and-forget: non-blocking analytics event
+  sendProductViewedAnalyticsEvent(streamableProduct).catch((err: unknown) => {
+    // eslint-disable-next-line no-console
+    console.error('Failed to send product viewed analytics event', err);
+  });
```

Comment:
Something we've been thinking about is handling this event being fired within the `with-routes` middleware. This way if a merchant statically caches the product pages, it can still be ran (because middleware isn't cached). While I think streaming this call in is a pretty clever idea, I think we can handling this on middleware will be more performant as streaming has some special performance qwirks.